src/test/typescript/typescript-parser_test.ts

```diff
@@ -0,0 +1,183 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {assert} from 'chai';
+// import * as fs from 'fs';
+// import * as path from 'path';
+import * as ts from 'typescript';
+
+import stripIndent = require('strip-indent');
+
+import {Analyzer} from '../../analyzer';
+import {UrlLoader} from '../../url-loader/url-loader';
+// import {FSUrlLoader} from '../../url-loader/fs-url-loader';
+import {UrlResolver} from '../../url-loader/url-resolver';
+import {ParsedTypeScriptDocument} from '../../typescript/typescript-document';
+import {TypeScriptParser} from '../../typescript/typescript-parser';
+
+class TestUrlResolver implements UrlResolver {
+  canResolve(url: string) {
+    return url === 'test.ts';
+  }
+
+  resolve(url: string) {
+    return (url === 'test.ts') ? '/typescript/test.ts' : url;
+  }
+}
+
+class TestUrlLoader implements UrlLoader {
+  canLoad(url: string) {
+    return url === '/typescript/test.ts';
+  }
+
+  load(url: string): Promise<string> {
+    if (url === '/typescript/test.ts') {
+      return Promise.resolve(`
+class A extends HTMLElement {
+  foo() { return 'bar'; }
+}
+`);
+    } else {
+      throw new Error(`cannot load file ${url}`);
+    }
+  }
+}
+
+suite('TypeScriptParser', () => {
+  let parser: TypeScriptParser;
+
+  setup(() => {
+    const urlLoader = new TestUrlLoader();
+    const urlResolver = new TestUrlResolver();
+    const analyzer = new Analyzer({urlLoader, urlResolver});
+    parser = new TypeScriptParser(analyzer);
+  });
+
+  suite('parse()', () => {
+
+    test('parses classes', () => {
+      const contents = `
+        import * as b from './b.ts';
+
+        class Foo extends HTMLElement {
+          bar: string = 'baz';
+        }
+      `;
+      const document = parser.parse(contents, '/typescript/test.ts');
+      assert.instanceOf(document, ParsedTypeScriptDocument);
+      assert.equal(document.url, '/typescript/test.ts');
+      console.log(document);
```

Comment:
Should clean up console logs and relevant ones to assertions (e.g. no diagnostics for a simple valid program)